src/main/java/uk/ac/ebi/pride/gui/task/AsperaGeneralTask.java

```diff
@@ -0,0 +1,217 @@
+package uk.ac.ebi.pride.gui.task;
+
+import com.asperasoft.faspmanager.FaspManager;
+import com.asperasoft.faspmanager.FaspManagerException;
+import com.asperasoft.faspmanager.InitializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.ebi.pride.App;
+import uk.ac.ebi.pride.data.exception.SubmissionFileException;
+import uk.ac.ebi.pride.data.io.SubmissionFileWriter;
+import uk.ac.ebi.pride.data.model.DataFile;
+import uk.ac.ebi.pride.gui.data.SubmissionRecord;
+import uk.ac.ebi.pride.gui.task.ftp.UploadErrorMessage;
+import uk.ac.ebi.pride.gui.task.ftp.UploadMessage;
+import uk.ac.ebi.pride.gui.task.ftp.UploadStoppedMessage;
+import uk.ac.ebi.pride.gui.util.Constant;
+import uk.ac.ebi.pride.gui.util.OSDetector;
+import uk.ac.ebi.pride.gui.util.SubmissionRecordSerializer;
+import uk.ac.ebi.pride.toolsuite.gui.desktop.DesktopContext;
+import uk.ac.ebi.pride.toolsuite.gui.task.TaskAdapter;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.security.SecureRandom;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * Parent class, handles most Aspera transfer general functionality.
+ */
+public abstract class AsperaGeneralTask extends TaskAdapter<Void, UploadMessage> {
+
+  /**
+   * Default constructor, initializes class variables.
+   */
+  private AsperaGeneralTask() {
+    this.filesToSubmit = Collections.synchronizedSet(new LinkedHashSet<File>());
+    this.totalFileSize = 0;
+  }
+
+  /**
+   * Constructor, initializes class variables including the submission record.
+   * @param submissionRecord the submission record to set
+   */
+  AsperaGeneralTask(SubmissionRecord submissionRecord) {
+    this();
+    this.submissionRecord = submissionRecord;
+  }
+
+  public static final Logger logger = LoggerFactory.getLogger(AsperaGeneralTask.class);
```

Comment:
I like to see the logger as the first line of the class as a convension

Reply:
In Following format:
Logger
Class Properties
Constructor(s)
Methods

Reply:
Fixed that now, I don't know who's bright idea it was to put class variables after the default class constructor.